### PR TITLE
[Merged by Bors] - chore(Topology/Homeomorph): rename type variables

### PR DESCRIPTION
**Pull request by grunweg**
X, Y, Z are standard mathematical names for topological spaces.
As [discussed](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Naming.20convention.3A.20topological.20spaces/near/395769893) on zulip, let us rename them.

### CODE REVIEW

**Comment by grunweg** [comment]
I don't quite understand the build errors and have asked [on Zulip](https://leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/Universe-related.28.3F.29.20error.20when.20renaming.20type.20variables). Help welcome!

**Comment by grunweg** [comment]
Found a fix, I think.

File: Mathlib/Topology/Homeomorph.lean [hunk 1]
```diff
@@ -437,253 +438,253 @@ def homeomorphOfContinuousOpen (e : α ≃ β) (h₁ : Continuous e) (h₂ : IsO
...
 
 variable {ι : Type*} {σ : ι → Type*} [∀ i, TopologicalSpace (σ i)]
 
-/-- `(Σ i, σ i) × β` is homeomorphic to `Σ i, (σ i × β)`. -/
-def sigmaProdDistrib : (Σi, σ i) × β ≃ₜ Σi, σ i × β :=
+/-- `(Σ i, σ i) × Y` is homeomorphic to `Σ i, (σ i × Y)`. -/
+def sigmaProdDistrib : (Σi, σ i) × Y ≃ₜ Σi, σ i × Y :=
   Homeomorph.symm <|
-    homeomorphOfContinuousOpen (Equiv.sigmaProdDistrib σ β).symm
+    homeomorphOfContinuousOpen (Equiv.sigmaProdDistrib σ Y).symm
```
Comment (alreadydone) [inline Mathlib/Topology/Homeomorph.lean:638]: I think these are the only Greek letters you missed. Would be reasonable to change to X.

**Review by alreadydone — APPROVED** [review]
Thanks!

**Review by alreadydone — APPROVED** [review]
Thanks!

File: Mathlib/Topology/Homeomorph.lean [hunk 2]
```diff
@@ -49,139 +49,140 @@ infixl:25 " ≃ₜ " => Homeomorph
...
-protected def trans (h₁ : α ≃ₜ β) (h₂ : β ≃ₜ γ) : α ≃ₜ γ where
+protected def trans (h₁ : X ≃ₜ Y) (h₂ : Y ≃ₜ Z) : X ≃ₜ Z where
   continuous_toFun := h₂.continuous_toFun.comp h₁.continuous_toFun
   continuous_invFun := h₁.continuous_invFun.comp h₂.continuous_invFun
   toEquiv := Equiv.trans h₁.toEquiv h₂.toEquiv
 #align homeomorph.trans Homeomorph.trans
 
 @[simp]
-theorem trans_apply (h₁ : α ≃ₜ β) (h₂ : β ≃ₜ γ) (a : α) : h₁.trans h₂ a = h₂ (h₁ a) :=
+theorem trans_apply (h₁ : X ≃ₜ Y) (h₂ : Y ≃ₜ Z) (a : X) : h₁.trans h₂ a = h₂ (h₁ a) :=
```
Comment (alreadydone) [inline Mathlib/Topology/Homeomorph.lean:121]: `x : X` would be appropriate

File: Mathlib/Topology/Homeomorph.lean [hunk 3]
```diff
@@ -49,139 +49,140 @@ infixl:25 " ≃ₜ " => Homeomorph
...
   rfl
 #align homeomorph.trans_apply Homeomorph.trans_apply
 
-@[simp] theorem symm_trans_apply (f : α ≃ₜ β) (g : β ≃ₜ γ) (a : γ) :
+@[simp] theorem symm_trans_apply (f : X ≃ₜ Y) (g : Y ≃ₜ Z) (a : Z) :
```
Comment (alreadydone) [inline Mathlib/Topology/Homeomorph.lean:125]: Here the original variable `a` didn't correspond to its type `γ`, but `z : Z` is probably better.

File: Mathlib/Topology/Homeomorph.lean [hunk 4]
```diff
@@ -742,31 +743,31 @@ end
...
-def toHomeomorph (e : α ≃ β) (he : ∀ s, IsOpen (e ⁻¹' s) ↔ IsOpen s) : α ≃ₜ β where
+def toHomeomorph (e : X ≃ Y) (he : ∀ s, IsOpen (e ⁻¹' s) ↔ IsOpen s) : X ≃ₜ Y where
   toEquiv := e
   continuous_toFun := continuous_def.2 λ s ↦ (he _).2
   continuous_invFun := continuous_def.2 λ s ↦ by convert (he _).1; simp
 
-@[simp] lemma coe_toHomeomorph (e : α ≃ β) (he) : ⇑(e.toHomeomorph he) = e := rfl
-lemma toHomeomorph_apply (e : α ≃ β) (he) (a : α) : e.toHomeomorph he a = e a := rfl
+@[simp] lemma coe_toHomeomorph (e : X ≃ Y) (he) : ⇑(e.toHomeomorph he) = e := rfl
+lemma toHomeomorph_apply (e : X ≃ Y) (he) (a : X) : e.toHomeomorph he a = e a := rfl
```
Comment (alreadydone) [inline Mathlib/Topology/Homeomorph.lean:756]: `x : X`

File: Mathlib/Topology/Homeomorph.lean [hunk 5]
```diff
@@ -49,139 +49,140 @@ infixl:25 " ≃ₜ " => Homeomorph
...
   left_inv := fun h => h.left_inv
   right_inv := fun h => h.right_inv
   coe_injective' := fun _ _ H _ => toEquiv_injective <| FunLike.ext' H
 
-instance : CoeFun (α ≃ₜ β) fun _ ↦ α → β := ⟨FunLike.coe⟩
+instance : CoeFun (X ≃ₜ Y) fun _ ↦ X → Y := ⟨FunLike.coe⟩
 
 @[simp]
-theorem homeomorph_mk_coe (a : Equiv α β) (b c) : (Homeomorph.mk a b c : α → β) = a :=
+theorem homeomorph_mk_coe (a : Equiv X Y) (b c) : (Homeomorph.mk a b c : X → Y) = a :=
```
Comment (alreadydone) [inline Mathlib/Topology/Homeomorph.lean:69]: Why was notation not used here for `Equiv`? Fixing it is in the scope of this PR, I think.

File: Mathlib/Topology/Homeomorph.lean [hunk 6]
```diff
@@ -49,139 +49,140 @@ infixl:25 " ≃ₜ " => Homeomorph
...
 #align homeomorph.trans_apply Homeomorph.trans_apply
 
-@[simp] theorem symm_trans_apply (f : α ≃ₜ β) (g : β ≃ₜ γ) (a : γ) :
+@[simp] theorem symm_trans_apply (f : X ≃ₜ Y) (g : Y ≃ₜ Z) (a : Z) :
     (f.trans g).symm a = f.symm (g.symm a) := rfl
 
 @[simp]
-theorem homeomorph_mk_coe_symm (a : Equiv α β) (b c) :
-    ((Homeomorph.mk a b c).symm : β → α) = a.symm :=
+theorem homeomorph_mk_coe_symm (a : Equiv X Y) (b c) :
```
Comment (alreadydone) [inline Mathlib/Topology/Homeomorph.lean:129]: and here

**Review by alreadydone — COMMENTED** [review]
Oh I saw the problems found by @eric-wieser in the other PR, so I took a closer look. Fortunately, there aren't much that could go wrong in this file ...

**Comment by grunweg** [comment]
Thank you for the fast review. I have addressed all comments (and also renamed an equivalence `a` to `h` twice).

File: Mathlib/Topology/Homeomorph.lean (hunk 5 above)
Comment (grunweg) [reply Mathlib/Topology/Homeomorph.lean:69]: Done. Also renamed `a` to `h`.

File: Mathlib/Topology/Homeomorph.lean (hunk 6 above)
Comment (grunweg) [reply Mathlib/Topology/Homeomorph.lean:129]: Same here.

**Comment by grunweg** [comment]
I'm puzzled by the build failures; I don't know enough to fix them. Help appreciated.

File: Mathlib/Topology/Homeomorph.lean [hunk 7]
```diff
@@ -49,139 +49,140 @@ infixl:25 " ≃ₜ " => Homeomorph
...
-@[simp] theorem symm_trans_apply (f : α ≃ₜ β) (g : β ≃ₜ γ) (a : γ) :
-    (f.trans g).symm a = f.symm (g.symm a) := rfl
+@[simp]
+theorem symm_trans_apply (f : X ≃ₜ Y) (g : Y ≃ₜ Z) (z : Z) :
+    (f.trans g).symm z = f.symm (g.symm z) := rfl
 
 @[simp]
-theorem homeomorph_mk_coe_symm (a : Equiv α β) (b c) :
-    ((Homeomorph.mk a b c).symm : β → α) = a.symm :=
+theorem homeomorph_mk_coe_symm (h : X ≃ₜ Y) (b c) :
```
[the remark this answers is not in the document]
Comment (alreadydone) [reply Mathlib/Topology/Homeomorph.lean:129]: Oh yeah, it should be `≃` not `≃ₜ`. What do you mean by "your version appears to be correct"?

[the remark this answers is not in the document]
Comment (adomani) [reply Mathlib/Topology/Homeomorph.lean:129]: Sorry, that was a very confusing remark: I was just trying to justify that, given all the previous assumptions, it looked like this one should have also been ≃ₜ, but actually the initial version was the intended one!

File: Mathlib/Topology/Homeomorph.lean [hunk 8]
```diff
@@ -49,139 +49,140 @@ infixl:25 " ≃ₜ " => Homeomorph
...
   left_inv := fun h => h.left_inv
   right_inv := fun h => h.right_inv
   coe_injective' := fun _ _ H _ => toEquiv_injective <| FunLike.ext' H
 
-instance : CoeFun (α ≃ₜ β) fun _ ↦ α → β := ⟨FunLike.coe⟩
+instance : CoeFun (X ≃ₜ Y) fun _ ↦ X → Y := ⟨FunLike.coe⟩
 
-@[simp]
-theorem homeomorph_mk_coe (a : Equiv α β) (b c) : (Homeomorph.mk a b c : α → β) = a :=
+@[simp] theorem homeomorph_mk_coe (a : X ≃ₜ Y) (b c) : (Homeomorph.mk a b c : X → Y) = a :=
```
Comment (alreadydone) [inline Mathlib/Topology/Homeomorph.lean:68]:
```suggestion
@[simp] theorem homeomorph_mk_coe (a : X ≃ Y) (b c) : (Homeomorph.mk a b c : X → Y) = a :=
```

File: Mathlib/Topology/Homeomorph.lean (hunk 7 above)
Comment (alreadydone) [inline Mathlib/Topology/Homeomorph.lean:129]:
```suggestion
theorem homeomorph_mk_coe_symm (h : X ≃ Y) (b c) :
```

**Comment by alreadydone** [comment]
maintainer merge

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Comment by grunweg** [comment]
@jcommelin I've merged master (to adapt to the changes in #7595. Can you approve again, please?

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**